calculator_test.go

```diff
@@ -24,31 +24,119 @@ func TestSubtract(t *testing.T) {
 }
 
 type TestCase struct {
-	A    float64
-	B    float64
+	name string
+	n    []float64
+	f    func(n ...float64) float64
 	want float64
 }
 
-func TestMultiply(t *testing.T) {
+func TestCalculator(t *testing.T) {
 	t.Parallel()
 	testCases := []TestCase{
 		{
-			A:    2,
-			B:    2,
-			want: 4,
+			name: "Add three positive integers",
+			n:    []float64{2, 2, 8},
+			f:    calculator.Add,
+			want: 12,
 		},
 		{
-			A:    -3,
-			B:    3,
-			want: -9,
+			name: "Add multiple negative and positive integers",
+			n:    []float64{-2, -2, 23, 40, 0},
+			f:    calculator.Add,
+			want: 59,
+		},
+		{
+			name: "Add a positive integer and a negative fraction ",
+			n:    []float64{5, -2.48},
+			f:    calculator.Add,
+			want: 2.52,
+		},
+		{
+			name: "Subtract a larger integer from a smaller integer",
+			n:    []float64{3, 5},
+			f:    calculator.Subtract,
+			want: -2,
+		},
+		{
+			name: "Subtract a negative integer from a negative integer",
+			n:    []float64{-9, -5, 1},
+			f:    calculator.Subtract,
+			want: -5,
+		},
+		{
+			name: "Multiply two positive integers",
+			n:    []float64{4, 20},
+			f:    calculator.Multiply,
+			want: 80,
+		},
+		{
+			name: "Multiply positive and a negative numbers",
+			n:    []float64{4, -9, 2},
+			f:    calculator.Multiply,
+			want: -72,
 		},
 	}
 
 	for _, tc := range testCases {
-		got := calculator.Multiply(tc.A, tc.B)
-		if tc.want != got {
-			t.Errorf("Multiply(%f, %f): want %f, got %f", 2.0, 2.0, tc.want, got)
-		}
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.f(tc.n...)
+			if tc.want != got {
```

Comment:
Floating-point values can be a bit awkward to compare, because they're inherently limited in precision. For example, 1/3 is 0.3333.... where the decimal expansion never stops. Similarly, there are numbers whose binary expansion is endless, and representing them within a fixed-size number of bits results in truncation.

So it's possible for a function to produce the right answer, but yet it doesn't quite compare equal. To be safe, we might have to instead compare two values by checking if the difference between is less than some small value.

Can you find some test cases where the results are correct yet don't compare equal? Make them pass by using a comparison function.